GetIntoTeachingApi/Services/CrmService.cs

```diff
@@ -87,20 +87,49 @@ public Candidate MatchCandidate(ExistingCandidateRequest request)
                 return null;
             }
 
-            _service.LoadProperty(entity, new Relationship("dfe_contact_dfe_candidatequalification_ContactId"), context);
-            _service.LoadProperty(entity, new Relationship("dfe_contact_dfe_candidatepastteachingposition_ContactId"), context);
-            _service.LoadProperty(entity, new Relationship("dfe_contact_dfe_servicesubscription_contact"), context);
-            _service.LoadProperty(entity, new Relationship("msevtmgt_contact_msevtmgt_eventregistration_Contact"), context);
+            LoadCandidateRelationships(entity, context);
 
             return new Candidate(entity, this);
         }
 
+        public IEnumerable<Candidate> MatchCandidates(string magicLinkToken)
+        {
+            // Avoids a potentially very expensive query.
+            if (string.IsNullOrEmpty(magicLinkToken))
+            {
+                return new Candidate[0];
```

Comment:
My reading of this is its an empty enumerable of Candidate models - have I got that right?

Reply:
Yeah exactly, we would never want to consider an empty token as a 'match' so returning an empty array here as 'no matches' - considered raising an exception instead but it just makes more logic to deal with that case in the same way further up the chain